rasa/nlu/test.py

```diff
@@ -201,16 +209,31 @@ def write_intent_successes(
         logger.info("No successful intent predictions found.")
 
 
-def write_intent_errors(
-    intent_results: List[IntentEvaluationResult], errors_filename: Text
-) -> None:
+def _write_errors(errors: List[Dict], errors_filename: Text, error_type: Text) -> None:
     """Write incorrect intent predictions to a file.
 
     Args:
-        intent_results: intent evaluation result
+        errors: Serializable prediction errors.
         errors_filename: filename of file to save incorrect predictions to
+        error_type: NLU entity which was evaluated (e.g. `intent` or `entity`).
     """
-    errors = [
+    if errors:
+        rasa.shared.utils.io.dump_obj_as_json_to_file(errors_filename, errors)
+        logger.info(f"Incorrect {error_type} predictions saved to {errors_filename}.")
+        logger.debug(
+            f"\n\nThese {error_type} examples could not be classified "
+            f"correctly: \n{errors}"
+        )
+    else:
+        if error_type.endswith("y"):
+            error_type_plural = f"{error_type[:-1]}ies"
```

Comment:
this seems a bit hacky, I've used https://pypi.org/project/inflect/ in the past.

Reply:
got rid of it by getting rid of the plural 😂